zh/monitoring-by-tidb-monitor.md

```diff
@@ -0,0 +1,208 @@
+```
+
+接下来我们可以将以下内容存为 yaml 文件，通过 kubectl apply -f 的方式部署部署一个 TidbMonitor 组件。
+
+> **注意：**
+>
+> 目前一个 TidbMonitor 只支持监控一个 TidbCluster。
+
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: TidbMonitor
+metadata:
+  name: basic
+spec:
+  clusters:
+  - name: basic
+  prometheus:
+    baseImage: prom/prometheus
+    version: v2.11.1
+    service:
+      type: NodePort
+  grafana:
+    baseImage: grafana/grafana
+    version: 6.0.1
+  initializer:
+    baseImage: pingcap/tidb-monitor-initializer
+    version: v3.0.5
+  reloader:
+    baseImage: pingcap/tidb-monitor-reloader
+    version: v1.0.1
+  imagePullPolicy: IfNotPresent
+```
+
+或者你可以通过以下方式快速部署 TidbMonitor:
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/basic/tidb-monitor.yaml -n <namespace>
+```
+
+然后我们通过 kubectl get pod 命令来检查 TidbMonitor 启动完毕:
+
+{{< copyable "shell-regular" >}}
+
+```shell
+$ kubectl get pod -l app.kubernetes.io/instance=basic | grep monitor 
+basic-monitor-85fcf66bc4-cwpcn     3/3     Running   0          117s
+```
+
+### 查看监控面板
+
+```shell
+> kubectl -n <namespace> port-forward svc/basic-grafana 4000:4000 &>/tmp/pf-grafana.log &
+```
+
+然后访问 localhost:4000
+
+### 删除集群与监控
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl delete tidbcluster basic -n <namespace>
+```
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl delete tidbmonitor basic -n <namespace>
+```
+
+## 持久化监控数据
+
+如果我们需要将 TidbMonitor 的监控数据持久化存储，我们需要在 TidbMonitor 中开启持久化选项:
+
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: TidbMonitor
+metadata:
+  name: basic
+spec:
+  clusters:
+    - name: basic
+  persistent: true
+  storageClassName: <storageClassName>
+  storage: 5G
+  prometheus:
+    baseImage: prom/prometheus
+    version: v2.11.1
+    service:
+      type: NodePort
+  grafana:
+    baseImage: grafana/grafana
+    version: 6.0.1
+    service:
+      type: NodePort
+  initializer:
+    baseImage: pingcap/tidb-monitor-initializer
+    version: v3.0.5
+  reloader:
+    baseImage: pingcap/tidb-monitor-reloader
+    version: v1.0.1
+  imagePullPolicy: IfNotPresent
+```
+
+你可以通过以下命令来确认 PVC 情况:
+
+```shell
+$ kubectl get pvc -l app.kubernetes.io/instance=basic,app.kubernetes.io/component=monitor
+NAME            STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+basic-monitor   Bound    pvc-6db79253-cc9e-4730-bbba-ba987c29db6f   5G         RWO            standard       51s
+```
+
+### 设置 kube-prometheus 与 AlertManager
+
+你可以通过设置 TidbMonitor 来使其获取 kube-prometheus metrics，了解 [kube-prometheus](https://github.com/coreos/kube-prometheus)
+
+同样的，你可以通过设置 TidbMonitor 来将监控推送警报至指定的 AlertManager，了解 [AlertManager](https://prometheus.io/docs/alerting/alertmanager/)
+
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: TidbMonitor
+metadata:
+  name: basic
+spec:
+  clusters:
+    - name: basic
+  kubePrometheusURL: "your-kube-prometheus-url"
```

Comment:
We'd better explain clearly what's the Prom used for here since we have a Prom in this deploy.

Reply:
updated.